setup.py

```diff
@@ -17,11 +17,16 @@
 import sys
 
 # Always prefer setuptools over distutils
-from setuptools import setup, find_packages
+from setuptools import setup, find_packages, __version__ as setuptoolsversion
 # To use a consistent encoding
 from codecs import open
 from os import path
 
+#require at least setuptools 21.0 for PEP 508 conditional dependency support
+MIN_SETUPTOOLS_VERSION = (21, 0)
+if tuple(int(x) for x in setuptoolsversion.split('.')[:2]) < MIN_SETUPTOOLS_VERSION:
```

Comment:
split has a [maxsplit](https://docs.python.org/2/library/stdtypes.html#str.split) argument you can use instead of indexing here

Reply:
@ppolewicz Unless I'm using it wrong, using maxsplit leaves the minor, patch, and any sub-patch versions unsplit.

```
>>> '21.1.0'.split('.',1)
['21', '1.0']
```

My goal here is to get the major and minor versions into a tuple for easy comparison, so it seems like I do need to do the slicing?

Reply:
oops, you are correct